pkg/scaffold/resource/options.go

```diff
@@ -0,0 +1,234 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resource
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/gobuffalo/flect"
+)
+
+const (
+	versionPattern = "^v\\d+(alpha\\d+|beta\\d+)?$"
+
+	groupRequired = "group cannot be empty"
+	versionRequired = "version cannot be empty"
+	kindRequired = "kind cannot be empty"
+)
+
+var versionRegex = regexp.MustCompile(versionPattern)
+
+// Options contains the information required to build a new Resource
+type Options struct {
+	// Namespaced is true if the resource is namespaced.
+	Namespaced bool
```

Comment:
add the bool at the end in order to use better the memory space. 
If you run the `maligned` you will check the error.